tensorflow/python/training/learning_rate_decay_test.py

```diff
@@ -108,5 +110,82 @@ def testPiecewiseConstantEdgeCases(self):
         pc = learning_rate_decay.piecewise_constant(x, boundaries, values)
 
 
+class ExponentialDecayTest(test_util.TensorFlowTestCase):
+
+  def testDecay(self):
+    initial_lr = 0.1
+    k = 10
+    decay_rate = 0.96
+    step = state_ops.variable_op([], dtypes.int32)
+    assign = state_ops.assign(step, 0)
+    add = state_ops.assign_add(step, 1)
+    with self.test_session():
+      assign.op.run()
+      for i in range(k+1):
+        decayed_lr = learning_rate_decay.natural_exp_decay(initial_lr, step,
```

Comment:
I still see this being called within the range loop, am I not seeing the latest version?